### ISSUE TITLE
How  do we run the test dataset in the temporal ConvNet?

### ISSUE THREAD

**Issue by rs096** [wadhwasahil/Video-Classification-2-Stream-CNN#11]
for chunk in chunks(keys,chunk_size):
			if flag<1:
				print("Preparing testing data...")
				X_test,Y_test=getTrainData(chunk,nb_classes,img_rows,img_cols)
				flag+=1
				continue
			print instance_count
			instance_count+=chunk_size
			X_batch,Y_batch=getTrainData(chunk,nb_classes,img_rows,img_cols)
			if (X_batch!=None and Y_batch!=None):
				loss = model.fit(X_batch, Y_batch, verbose=1, batch_size=batch_size, nb_epoch=1, show_accuracy=True)	
				if instance_count%256==0:
					loss = model.evaluate(X_test,Y_test,batch_size=batch_size,verbose=1)
					preds = model.predict(X_test)
					print (preds)
					print ('-'*40)
					print (Y_test)
					comparisons=[]
					maximum=np.argmax(Y_test,axis=1)
					for i,j in enumerate(maximum):
						comparisons.append(preds[i][j])
					with open('compare.txt','a') as f1:
						f1.write(str(comparisons))
						f1.write('\n\n')
					with open('loss.txt','a') as f1:
						f1.write(str(loss))
						f1.write('\n')
					model.save_weights('temporal_stream_model.h5',overwrite=True)

So is the code which running the testing part of the code ?

**Comment by wadhwasahil** [comment]
yes, training and testing after some(predefined) epochs.

**Comment by rs096** [comment]
Using TensorFlow backend.
Loading dictionary...
Preparing architecture...
/home/user/PycharmProjects/Video-Classification-2-Stream-CNN-master/temporal/temporal_stream_cnn.py:54: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (7, 7), padding="same", input_shape=(20, 150, ...)`
  model.add(Convolution2D(48, 7, 7, border_mode='same',input_shape=(img_channels, img_rows, img_cols)))
/home/user/PycharmProjects/Video-Classification-2-Stream-CNN-master/temporal/temporal_stream_cnn.py:59: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(96, (5, 5), padding="same")`
  model.add(Convolution2D(96, 5, 5, border_mode='same'))
/home/user/PycharmProjects/Video-Classification-2-Stream-CNN-master/temporal/temporal_stream_cnn.py:64: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(256, (3, 3), padding="same")`
  model.add(Convolution2D(256, 3, 3, border_mode='same'))
/home/user/PycharmProjects/Video-Classification-2-Stream-CNN-master/temporal/temporal_stream_cnn.py:68: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(512, (3, 3), padding="same")`
  model.add(Convolution2D(512, 3, 3, border_mode='same'))
/home/user/PycharmProjects/Video-Classification-2-Stream-CNN-master/temporal/temporal_stream_cnn.py:72: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(512, (3, 3), padding="same")`
  model.add(Convolution2D(512, 3, 3, border_mode='same'))
Compiling model...
----------------------------------------
('Epoch', 0)
----------------------------------------
Preparing testing data...
0
64
128
192
256
320
384

so is this the testing part or this is just collection of testing data

**Comment by wadhwasahil** [comment]
Yes this is the test data been prepared. It is just the number of videos processed/prepared for testing AFAIK.

**Comment by rs096** [comment]
So when will it start predicting the action in the video , since I can't
find the part within the code which will actually classify test video into
anyone of 20 categories.

**Comment by wadhwasahil** [comment]
preds = model.predict(X_test) does the predictions of test videos.

**Comment by rs096** [comment]
model.add(Convolution2D(512, 3, 3, border_mode='same'))
Compiling model...
----------------------------------------
('Epoch', 0)
----------------------------------------
Preparing testing data...
0
64
128
192
256
320
384
......
('epoch',20)
.....
.
...
178048
178112
178176
178240
178304
178368
178432
178496
178560
178624
178688
178752

Process finished with exit code 0


so that is it this is showing where do I recieve the final prediced classes and stuff since it is not shown in the out
